### PR TITLE
Change agents_position to a list and add a wrapper to observations

### PR DESCRIPTION
**Pull request by RicardoRibeiroRodrigues**

### CODE REVIEW

File: DSSE/environment/env_base.py [hunk 1]
```diff
@@ -167,11 +166,11 @@ def is_valid_position(self, position: tuple[int, int]) -> bool:
     def default_drones_positions(self):
         counter_x = 0
         counter_y = 0
-        for agent in self.agents:
+        for agent_index in range(len(self.agents)):
```
Comment (renatex333) [inline DSSE/environment/env_base.py:169]: Maybe use enumerate instead of `range(len())`

File: DSSE/environment/wrappers/all_positions_wrapper.py
Comment (renatex333) [inline DSSE/environment/wrappers/all_positions_wrapper.py:1]: Is this work in progress? I saw some hardcoded variables, I'm not sure if they are inteded

**Review by renatex333 — APPROVED** [review]

File: DSSE/environment/env_base.py (hunk 1 above)
Comment (RicardoRibeiroRodrigues) [reply DSSE/environment/env_base.py:169]: We only need the index on this loop, i don't see a reason.

Comment (renatex333) [reply DSSE/environment/env_base.py:169]: Fair enough 😂

File: DSSE/environment/wrappers/all_positions_wrapper.py
Comment (renatex333) [reply DSSE/environment/wrappers/all_positions_wrapper.py:1]: In add_other_positions_obs, I saw that you accessed the prob_matrix from the "drone0", which I thought, at first, could be just for testing, but I understand now that you just needed to get the matrix. I was trying to think of a more elegant way to do that, but I'm not sure there is one. Anyway, thank you for your atention!

**Review by JorasOliveira — APPROVED** [review]

**Pull request merged**